src/rpc/mempool.cpp

```diff
@@ -0,0 +1,206 @@
+            "\nThis checks if the transaction violates the consensus or policy rules.\n"
+            "\nSee sendrawtransaction call.\n"
+            "\nArguments:\n"
+            "1. [\"rawtxs\"]        (array, required) An array of hex strings of raw transactions.\n"
+            "2. allowhighfees    (boolean, optional, default=false) Allow high fees\n"
+            "\nResult:\n"
+            "  \"allowed\"            (boolean) If the mempool allows this tx to be inserted\n"
+            "  \"reject-reason\"  (string) Rejection string (only present when 'allowed' is false)\n"
+            "[                   (array) The result of the mempool acceptance test for each raw transaction in the input array. Length is equal to the number of input transactions."
+            " {\n"
+            "  \"<txid>:\"                  (string) The transaction id in hex\n"
+            "   {\n"
+            "    \"allowed\"               (boolean) If the mempool allows this tx to be inserted\n"
+            "    \"reject-reason\"     (string) Rejection string (only present when 'allowed' is false)\n"
+            "   }\n"
+            " }\n"
+            "]\n"
+            "\nExamples:\n"
+            "\nCreate a transaction\n"
+            + HelpExampleCli("createrawtransaction", "\"[{\\\"txid\\\" : \\\"mytxid\\\",\\\"vout\\\":0}]\" \"{\\\"myaddress\\\":0.01}\"") +
+            "Sign the transaction, and get back the hex\n"
+            + HelpExampleCli("signrawtransaction", "\"myhex\"") +
+            "\nTest acceptance of the transaction (signed hex)\n"
+            + HelpExampleCli("testmempoolaccept", "\"signedhex\"") +
+            "\nAs a json rpc call\n"
+            + HelpExampleRpc("testmempoolaccept", "[\"signedhex\"]")
+            // clang-format on
+            );
+    }
+
+    RPCTypeCheck(request.params, {UniValue::VARR, UniValue::VBOOL});
+    const UniValue raw_transactions = request.params[0].get_array();
+    if (raw_transactions.size() < 1 || raw_transactions.size() > MAX_PACKAGE_COUNT) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER,
+                            "Too many transactions in package." );
+    }
+
+    Package package;
+    for (const auto& rawtx : raw_transactions.getValues()) {
+        CMutableTransaction mtx;
+        if (!DecodeHexTx(mtx, rawtx.get_str())) {
+            throw JSONRPCError(RPC_DESERIALIZATION_ERROR,
+                                "TX decode failed: " + rawtx.get_str() + " Make sure the tx has at least one input.");
+        }
+        package.push_back(MakeTransactionRef(std::move(mtx)));
+    }
+
+    if(package.size() != raw_transactions.size()){
+            throw JSONRPCError(RPC_INVALID_PARAMETER,
+                    "Duplicate transaction found in package.");
+    }
+
+    CAmount max_raw_tx_fee = ::maxTxFee;
+    if (!request.params[1].isNull() && request.params[1].get_bool()) {
+        max_raw_tx_fee = 0;
```

Comment:
This variable doesn't seem to be used after this, how does it work?

Reply:
This was a bug! The option should be used to allow high fee transactions. 
It is now fixed by setting a flag in `CTxMempoolAcceptanceOptions` options in `AcceptToMemoryPoolWorker`.